### ISSUE TITLE
Home Screen Module Logic - Claims

### ISSUE THREAD

**Issue by ajsarkar28** [department-of-veterans-affairs/va-mobile-app#7378]
**Background**
* This ticket covers the content and display logic for the Claims module which displays on the home screen

**Acceptance Criteria**
* If the Veteran has open Claims, based on the backend work in the ticket in the comments, then display a module on the home screen with the following content:
    * Text that says “Claims (x open)” 
        * In this example, x should be replaced with the number of unread Claims returned from the backend. For instance, if the Veteran has 2 open Claims, the module will read “Claims (x open)” 
    * If clicked, the module should deep link the user to the Claims History page
 * If the Veteran has 0 Claims returned from the backend, then do NOT display a Claims module
 * NOTE - error handling is not finalized and will be covered in a separate ticket (ex. health) not feature level (ex. just prescriptions)
 * NOTE - This ticket is strictly display logic. The UI will be covered in a separate ticket once designs are finalized

**Comment by ajsarkar28** [comment]
https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/7213

**Comment by TKDickson** [comment]
Not an official blocking relationship, but the 'ugly count' work in this ticket will only reflect claims (not appeals) until #7468 is done.

Or rather - it wouldn't block FE, but does block QA.

**Comment by TKDickson** [comment]
QA estimate (once the BE ticket is done) - 2 points (because the data will inevitably fall apart for this).

**Comment by TKDickson** [comment]
Testing
- [x]  User with ClaimCount of 1 or more (mock it if needed) - module shows up on the homescreen, with the correct # displayed
- [x]  User with ClaimCount of 0 - the module does not appear on the homescreen


For a user when the module appears on the screen:
- [x]   when I tap on the module, it navigates to the Claims and Appeals list
- [x]  when I tap '< Benefits' at this point, it navigates to Benefits category landing
- [x]  or if I tap on the built-in back (Android back button, iOS swipe), it navigates to the Benefits category landing

**Comment by TKDickson** [comment]
@theodur for a user who doesn't have access to decision letters (either from VA, or via a Waygate), but does have active claims, we'll need to add something for navigation I think. 

What I'm seeing (for a mocked up scenario of this with +54, since I don't know of any staging users) is:
* Use home screen to jump into list of claims
* Go back -- which I'd expect would go straight to the Benefits landing -- and am brought to 'Claims History / Claim Letters' screen
* At this point I can tap the Claim Letters menu option, and (for a real user) I imagine it would always error out.

**Comment by theodur** [comment]
@TKDickson It looks like we have a check on the Claim letters screen that will display the no letters component when a user doesn't have any letters, which also seems to apply if a user doesn't have access (I mocked the authorized services endpoint locally). Here's what that looks like:

<img width="548" alt="image" src="https://github.com/department-of-veterans-affairs/va-mobile-app/assets/9042882/8abbbdc2-f83f-48d8-b3db-771b5189d500">

As for navigating back to the Benefits screen after deep linking to the Claims history screen, this would require us to dynamically set the label and functionality of the back button for that screen, which could be a bit cumbersome. 

My concern from a UX perspective is that users who don't have access to decision letters are still able to access the 'Claims history/ Claim letters' screen when manually navigating through the app, and the default behavior when manually navigating to the Claims history screen and tapping the back button takes the user back to the 'Claims history/ Claim letters' screen. So changing the back navigation when a deep link takes a user to the Claims history screen could be confusing for users since that would be inconsistent with the behavior when manually navigating to the screen. I'm not sure that UX specified custom back navigation for this module, but maybe we can confirm with them

**Comment by TKDickson** [comment]
@theodur I was experiencing that a user without access to decision letters *can't* manually navigate to that 'history/letters' screen, they're just skipped over it straight to the claims list. This from BenefitsScreen.tsx:

  ```
const onClaims = () => {
    if (featureEnabled('decisionLettersWaygate') && userAuthorizedServices?.decisionLetters) {
      navigateTo('Claims')()
    } else if (waygateNativeAlert('WG_ClaimsHistory')) {
      navigateTo('ClaimsHistory')()
    }
  }
```

I think checking with the UX team is a good idea, I'll bring it up as either an 11th minute topic today, or drop something in Slack.

**Comment by theodur** [comment]
@TKDickson Ohhh shoot, the reason why I was able to manually navigate to the 'history/letters' screen is because I was mocking the user not having access to `lettersAndDocuments` instead of `decisionLetters` 🤦 my bad! This makes a lot more sense now. I definitely think that if a user isn't able to manually navigate to that screen, the back navigation should reflect that when deep linking. I'll update my PR today to handle navigating back to the Benefits screen when the user doesn't have access to `decisionLetters`.

**Comment by theodur** [comment]
@TKDickson The issue should be fixed now ✅ let me know if you run into any other issues

**Comment by TKDickson** [comment]
Confirmed fixed! With that, all cases pass. This is ready to merge into XL.

**Comment by TKDickson** [comment]
Closing, merged to XL.
